api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -200,111 +275,104 @@ def publish_cdr(cmd_name, args)
+  dest_dataset = "#{tier.fetch(:dest_cdr_project)}:#{op.opts.bq_dataset}"
+
+  service_account_context_for_bq(op.opts.project, env.fetch(:publisher_account)) do
+    bq_ingest(tier, source_cdr_project, op.opts.bq_dataset, table_match_filter, table_skip_filter)
+
+    bq_update_acl(dest_dataset) do |acl_json, existing_groups, existing_users|
+      auth_domain_group_email = tier.fetch(:auth_domain_group_email)
+      if existing_groups.include?(auth_domain_group_email)
+        common.status "#{auth_domain_group_email} already in ACL, skipping..."
+      else
+        common.status "Adding #{auth_domain_group_email} as a READER..."
+        acl_json["access"].push({
+          "groupByEmail" => auth_domain_group_email,
+          "role" => "READER"
+        })
       end
 
-      unless op.opts.exclude_auth_domain_acl
-        if existing_groups.include?(auth_domain_group_email)
-          common.status "#{auth_domain_group_email} already in ACL, skipping..."
-        else
-          common.status "Adding #{auth_domain_group_email} as a READER..."
-          new_entry = { "groupByEmail" => auth_domain_group_email, "role" => "READER"}
-          json["access"].push(new_entry)
-        end
+      app_sa = "#{op.opts.project}@appspot.gserviceaccount.com"
+      if existing_users.include?(app_sa)
+        common.status "#{app_sa} already in ACL, skipping..."
+      else
+        common.status "Adding #{app_sa} as a READER..."
+        acl_json["access"].push({ "userByEmail" => app_sa, "role" => "READER"})
       end
 
-      # if the app SA's in too many groups, it won't gain READER transitively.
-      # add it directly, to make sure.
-      # See discussion at https://pmi-engteam.slack.com/archives/CHRN2R51N/p1609869521078200?thread_ts=1609796171.063800&cid=CHRN2R51N
-      unless op.opts.exclude_sa_acl
-        if existing_users.include?(app_sa)
-          common.status "#{app_sa} already in ACL, skipping..."
-        else
-          common.status "Adding #{app_sa} as a READER..."
-          new_entry = { "userByEmail" => app_sa, "role" => "READER"}
-          json["access"].push(new_entry)
-        end
-      end
+      return acl_json
+    end
+  end
+end
 
-      if op.opts.additional_reader_group
-        new_group = op.opts.additional_reader_group
+Common.register_command({
+  :invocation => "publish-cdr",
+  :description => "Publishes a CDR dataset by copying it into a Firecloud CDR project and making it readable by registered users in the corresponding environment",
+  :fn => ->(*args) { publish_cdr("publish-cdr", args) }
+})
 
-        if existing_groups.include?(new_group)
-          common.status "#{new_group} already in ACL, skipping..."
-        else
-          common.status "Adding #{new_group} as a READER..."
-          new_entry = { "groupByEmail" => new_group, "role" => "READER"}
-          json["access"].push(new_entry)
-        end
-      end
+def publish_cdr_wgs(cmd_name, args)
```

Comment:
FYI: I've disabled this pattern, it was previously disabled in Codacy per https://docs.google.com/document/d/1o1nN4Oni9v6wgoYeFhG2yj47pqaD7XkVrRaTHqsdHhg/edit#heading=h.2f4f7hkj818a